### PR TITLE
Refactor/init lombok

### PR DESCRIPTION
**Pull request by marcambier**
https://github.com/gravitee-io/issues/issues/6827

**Description**

This adds lombok dependency to APIM rest API ; and use it in a first data object.

The idea is to gradually migrate APIM data objects using lombok, when we work on them.
In order to reduce code boilerplate.

🚀 CI was able to deploy the build of this PR, so you can now try it directly [here](https://apimnightlywebui24386.blob.core.windows.net/refactor-initlombok/index.html)
_Notes_: The deployed app is linked to the management API of APIM master. (Same login and password as APIM master)

### CODE REVIEW

File: gravitee-apim-rest-api/gravitee-apim-rest-api-model/src/main/java/io/gravitee/rest/api/model/ApplicationEntity.java [hunk 1]
```diff
@@ -21,12 +21,18 @@
...
+import lombok.ToString;
 
 /**
  * @author David BRASSELY (david.brassely at graviteesource.com)
  * @author Nicolas GERAUD (nicolas.geraud at graviteesource.com)
  * @author GraviteeSource Team
  */
+@Getter
+@Setter
+@ToString
```
Comment (ytvnr) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-model/src/main/java/io/gravitee/rest/api/model/ApplicationEntity.java:35]: do we want to keep our equals and hashcode ?
if not, we can maybe use `@Data` ?

**Review by ytvnr — APPROVED** [review]
lgtm

File: gravitee-apim-rest-api/gravitee-apim-rest-api-model/src/main/java/io/gravitee/rest/api/model/ApplicationEntity.java (hunk 1 above)
Comment (marcambier) [reply gravitee-apim-rest-api/gravitee-apim-rest-api-model/src/main/java/io/gravitee/rest/api/model/ApplicationEntity.java:35]: Lombok-generated equals would compare all properties by default.

We could make it compare only ID, using this annotation on class : `@EqualsAndHashCode(onlyExplicitlyIncluded = true)`
And this one on`id`property : @EqualsAndHashCode.Include

WDYT ? Would it more more explicit/comprehensible with or without lombok for this case ?

Comment (marcambier) [reply gravitee-apim-rest-api/gravitee-apim-rest-api-model/src/main/java/io/gravitee/rest/api/model/ApplicationEntity.java:35]: As discussed together, I added that

**Pull request merged**